server/mneme_api_response.rb

```diff
@@ -80,6 +81,49 @@ def extractAssignment(assignment)
     assign
   end
 
+  ############ Filter assignments
+  # The filter will take a single assignment and return it if it is acceptable and return nil
+  # if the assignment should not be used.
+
+  # The checks require date comparisons. To make testing easier the actual filter is implemented as
+  # a class method which requires the time as an argument.  The following instance method defaults
+  # the time to the current time to now.  'now' is a method that can be overridden by tests.
+
+  # This is an instance method to bind the time used for filtering to the current time.
+  def filter(assignment)
+    result = self.class.filter_two(assignment, Time.now().to_i)
+    logger.debug "#{self.class.to_s}:#{__method__}: #{__LINE__}: skipping assignment: [#{assignment.inspect}]" unless result
+    return result
+  end
+
+  # This is a class method for easy testing.  The current time must be passed in
+  def self.filter_two(assignment, now)
```

Comment:
Why the name "filter_two"?  A name like "isOpenPublishedAssignment" would be better.

Does the current time really need to be passed in?  If this will always use the current time, call `Time.now().to_i` in this method so that the argument list can be shortened.